### PR TITLE
Twig function to generate link to a filtered page

### PR DESCRIPTION
**Pull request by QuentinHAETTEL**

### CODE REVIEW

File: doc/twig.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+```twig
+{{ crudit_route_filtered_link('entity', [['field1', 'value1', 'operator1'], ['field2', {'key2.1': 'value2.1', 'key2.2': 'value2.2'}, 'operator2']]) }}
+```
+
+The operator is not required, default is "eq".
+
+Example :
+```twig
+{{ crudit_route_filtered_link('user', [['status', 'generated'], ['active', {'value': '2023-01-01', 'to': '2023-12-31'}, 'interval']]) }}
```
Comment (SpaghettiBolognaise) [inline doc/twig.md:17]: i don't like mixing values types.

it should be a simple and unified format, nothing else

like this: { 'status': { 'value': ...', 'op': ..., 'items': ...}}

(op being optional and items would work for entityfilter)

File: src/Twig/RouteFilteredLinkExtension.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        ];
+    }
+
+    public function getRouteFilteredLink(string $entity, array $filters)
+    {
+        $route = 'app_crudit_' . $entity . '_index';
+
+        try {
+            $this->router->generate($route);
+        } catch (RouteNotFoundException $e) {
```
Comment (SpaghettiBolognaise) [inline src/Twig/RouteFilteredLinkExtension.php:33]: don't catch that error, the dev has to fix it or not use it...

File: src/Twig/RouteFilteredLinkExtension.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+            if (is_array($filter[1])) {
+                foreach ($filter[1] as $key => $value) {
+                    $parameters[$filterName . '_' . $key] = $value;
+                }
+            } else {
+                $parameters[$filterName . '_value'] = $filter[1];
+            }
+        }
```
Comment (SpaghettiBolognaise) [inline src/Twig/RouteFilteredLinkExtension.php:54]: refactor that and avoid using raw numeric array keys

File: src/Twig/RouteFilteredLinkExtension.php (hunk 2 above)
Comment (sebheitzmann) [reply src/Twig/RouteFilteredLinkExtension.php:33]: d'accord

**Review by SpaghettiBolognaise — APPROVED** [review]

**Pull request merged**